src/ape/types/__init__.py

```diff
@@ -264,6 +267,19 @@ def get(self, item: str, default: Optional[Any] = None) -> Any:
         return self.event_arguments.get(item, default)
 
 
+class ContractLogContainer(list):
+    __test__ = False
+
+    def filter(self, event, **kwargs) -> List[ContractLog]:
```

Comment:
Needs a docstrings and typing

Reply:
if i give type for event it would be ContractEvent and .contracts.base importing ContractLog then this is a import cycle. and i cant just use "ContractEvent" if i dont import it some point and kwargs i dont have any type basically, but will add docstring

Reply:
Could you do:

```python
from typing import TYPE_CHECKING
if TYPE_CHECKING:
    from ape.contracts.base import ContractEvent

...

def filter(self, event: "ContractEvent", **kwargs):

Reply:
Yeah, you don't need a type for kwargs, although I think `Dict[str, Any] ` might work

Reply:
BTW for import cycles you can solve that using quotes around your type

Reply:
Types on `**kwargs` can be a little unintuitive. I agree, don't worry about them; we don't.

But, just so you know, for `*args` and `**kwargs`, you can only put the type of the value(s), like `*args: Any` or `**kwargs: Any`, you don't do `*args: List[Any]` or `**kwargs: Dict[str, Any]`.

Reply:
yes, i found this example in code regarding TYPE_CHECKING so it is ok now.